opentracing-examples/src/test/java/io/opentracing/examples/common_request_handler/TestHandler.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2016-2017 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.examples.common_request_handler;
+
+import io.opentracing.ActiveSpan;
+import io.opentracing.mock.MockSpan;
+import io.opentracing.mock.MockTracer;
+import io.opentracing.mock.MockTracer.Propagator;
+import io.opentracing.tag.Tags;
+import io.opentracing.util.ThreadLocalActiveSpanSource;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+import static io.opentracing.examples.TestUtils.sortByStartMicros;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+/**
+ * There is only one instance of 'RequestHandler' per 'Client'. Methods of 'RequestHandler' are
+ * executed concurrently in different threads which are reused (common pool). Therefore we cannot
+ * use current active span and activate span. So one issue here is setting correct parent span.
+ */
+public class TestHandler {
+
+    private final MockTracer tracer = new MockTracer(new ThreadLocalActiveSpanSource(),
+            Propagator.TEXT_MAP);
+    private final Client client = new Client(new RequestHandler(tracer));
+
+    @Before
+    public void before() {
+        tracer.reset();
+    }
+
+    @Test
+    public void two_requests() throws Exception {
+        Future<Object> responseFuture = client.send("message");
+        Future<Object> responseFuture2 = client.send("message2");
+
+        assertEquals("message:response", responseFuture.get(15, TimeUnit.SECONDS));
+        assertEquals("message2:response", responseFuture2.get(15, TimeUnit.SECONDS));
+
+        List<MockSpan> finished = tracer.finishedSpans();
+        assertEquals(2, finished.size());
+
+        for (MockSpan span : finished) {
+            assertEquals(Tags.SPAN_KIND_CLIENT, span.tags().get(Tags.SPAN_KIND.getKey()));
+        }
+
+        assertNotEquals(finished.get(0).context().traceId(), finished.get(1).context().traceId());
+        assertEquals(finished.get(0).parentId(), finished.get(1).parentId());
```

Comment:
I think that these two last lines are distracting. I would rather see asset on `0` it's more explicit.